### ISSUE TITLE
Serious Bug: Admin/Frontend Template Path Hints and Add Block Names to Hits no longer functioning

### ISSUE THREAD

**Issue by ADDISON74** [OpenMage/magento-lts#1413]
OpenMage version 20.0.6. 

Go to **Backend -> System -> Configuration -> Developer -> Debug**

**STEP 1**: From the left sidebar list [Current Configuration Scope] select the first option Default Config. Now set one by one as follows:

- Admin Template Path Hints: Yes. By pressing [Save Config] you see the correct effects immediately.
- Admin Add Block Names to Hints: Yes. By pressing [Save Config] you see the correct effects immediately.

- Frontend Template Path Hints: Yes. By pressing [Save Config] **no frontend effects**.
- Frontend Add Block Names to Hints: Yes. By pressing [Save Config] **no frontend effects**.

**STEP 2**: Restore the default values we changed in step 1 to No. From the left sidebar list [Current Configuration Scope] select the second option Main Website. Now set one by one as follows:

- Admin Template Path Hints: Uncheck "Use Default" and select Yes. By pressing [Save Config] no backend effect and there is an error message "Wrong field specified". Selected option remains set to No.

The same results for the other options "Admin Add Block Names to Hints", "Frontend Template Path Hints", "Frontend Add Block Names to Hints". Any of these values cannot be set to Yes. See the image bellow.

![screenshot](https://user-images.githubusercontent.com/8360474/105478869-6bf0b700-5cac-11eb-8f27-3e310e4a3dd6.jpg)

I overwrite all files with OpenMage 20.0.5 and I did not find any issues. It worked as expected just see bellow the configuration was saved correctly.

![screensho2](https://user-images.githubusercontent.com/8360474/105480021-e3731600-5cad-11eb-9be5-f95f5eb38c74.jpg)

In conclusion this is a serious bug which affect developers: 

1. "Something" bad happening between 20.0.5 to 20.0.6 related to path hints. Splitting them in two sections was a good idea but the code behind creates issues.

2. I am no longer able to use template and block names hints in frontend. A solid fix should be release ASAP.

3. I cannot set Yes values for different configuration scopes like Main Website, Default Store View.

**Comment by ADDISON74** [comment]
I managed to solve this issue. Templates Hints are working again as expected.

Here is the solution I used: https://github.com/OpenMage/magento-lts/issues/1417#issuecomment-765710454

**Comment by ADDISON74** [comment]
Solved based on the solution @colinmollenhour provided here #1418. I will close this issue.
